pymc/smc/sampling.py

```diff
@@ -236,20 +238,28 @@ def sample_smc(
     )
 
     if compute_convergence_checks:
-        _compute_convergence_checks(idata, draws, model, trace)
-    return idata if return_inferencedata else trace
+        if idata is None:
+            idata = to_inference_data(trace, log_likelihood=False)
+        warns = run_convergence_checks(idata, model)
+        trace.report._add_warnings(warns)
+        log_warnings(warns)
```

Comment:
This replaces the `_compute_convergence_checks` function and makes the `trace.report` be a dead end that can easily be removed in the future

Remember from other changes:
* "number of samples is too small" warning now done by `run_convergence_checks`
* `report._add_warnings` was done inside `report._run_convergence_checks`
* `trace.report._log_summary()` internally called `log_warnings()`